scripts/prune_lm_dir.py

```diff
@@ -21,6 +22,15 @@
                     'prune*X, with X <= 1.0, tells it to prune with X times the threshold '
                     'specified with the --threshold option.  EM specifies one iteration of '
                     'E-M on the model. ')
+parser.add_argument("--target-size", type=int, default=0,
+                    help="Target size of final LM after pruning. "
+                    "If setting this to a positive value, the --steps would be "
+                    "ignored and a few steps may be worked out util the size "
+                    "of pruned LM match the target-size.")
+parser.add_argument("--tolerance", type=float, default=0.05,
+                    help="Tolerance of actual size of final LM and the target-size. ")
+parser.add_argument("--max-attempt", type=int, default=20,
+                    help="Max attempts to find target size LM. ")
```

Comment:
Yes, I think that would be best.

On Tue, Jul 26, 2016 at 8:05 PM, Wang Jian <notifications@github.com> wrote:

> In scripts/prune_lm_dir.py
> <https://github.com/danpovey/pocolm/pull/31#discussion_r72374274>:
>
> > @@ -21,6 +22,15 @@
> >                      'prune*X, with X <= 1.0, tells it to prune with X times the threshold '
> >                      'specified with the --threshold option.  EM specifies one iteration of '
> >                      'E-M on the model. ')
> > +parser.add_argument("--target-size", type=int, default=0,
> > +                    help="Target size of final LM after pruning. "
> > +                    "If setting this to a positive value, the --steps would be "
> > +                    "ignored and a few steps may be worked out util the size "
> > +                    "of pruned LM match the target-size.")
> > +parser.add_argument("--tolerance", type=float, default=0.05,
> > +                    help="Tolerance of actual size of final LM and the target-size. ")
> > +parser.add_argument("--max-attempt", type=int, default=20,
> > +                    help="Max attempts to find target size LM. ")
>
> OK, is it necessary to change the working dir from work/iterX to
> work/stepX?
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/danpovey/pocolm/pull/31/files/976e873ceae43f66711fce0e4e76e9471d773312#r72374274>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ADJVu9TDdPtAffM2OlNJdAvRcFxPt__9ks5qZssVgaJpZM4JU463>
> .
>